centos/common/network-config.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -e
+
+sed -i '/\[main\]/a no-auto-default=*' /etc/NetworkManager/NetworkManager.conf
+
+# update network config on reboot
+mkdir -p /lib/systemd/system/cloud-init-local.service.d/
+cat <<EOF > /lib/systemd/system/cloud-init-local.service.d/50-azure-clear-persistent-obj-pkl.conf
+[Service]
+ExecStartPre=-/bin/sh -xc 'if [ -e /var/lib/cloud/instance/obj.pkl ]; then echo "cleaning persistent cloud-init object"; rm /var/lib/cloud/instance/obj.pkl; fi; exit 0'
+EOF
```

Comment:
Also, please make sure there are no side effects with AN.

Reply:
Tested with Hc with AN enabled, disabled - both showed benefit. No reason to think that AN will impact.